### PR TITLE
[MI-3644] Add metrics of subscription count for different actions

### PR DESCRIPTION
**Pull request by ayusht2810**
#### Summary
Add metrics of subscription count for different actions

### CODE REVIEW

**Review by raghavaggarwal2308 — APPROVED** [review]

File: server/metrics/metrics.go [hunk 1]
```diff
@@ -188,6 +190,15 @@ func NewMetrics(info InstanceInfo) Metrics {
 	}, []string{"source", "is_direct"})
 	m.registry.MustRegister(m.messagesConfirmedCount)
 
+	m.subscriptionsCount = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace:   MetricsNamespace,
+		Subsystem:   MetricsSubsystemEvents,
+		Name:        "subscriptions_count",
+		Help:        "The total number of subscriptions connected, reconnected and refreshed.",
```
Comment (Kshitij-Katiyar) [inline server/metrics/metrics.go:197]: Maybe ?
```suggestion
		Help:        "The total number of connected, reconnected and refreshed subscriptions.",
```

File: server/plugin.go [hunk 2]
```diff
@@ -53,6 +53,8 @@ const (
 	actionDeleted          = "deleted"
 	reactionSetAction      = "set"
 	reactionUnsetAction    = "unset"
+	subscriptionConnected  = "connected"
```
Comment (Kshitij-Katiyar) [inline server/plugin.go:56]: This constant is already defined i think

Comment (ayusht2810) [reply server/plugin.go:56]: same https://github.com/mattermost/mattermost-plugin-msteams-sync/pull/354#discussion_r1368341106

**Review by Kshitij-Katiyar — APPROVED** [review]

**Review by lieut-data — APPROVED** [review]
Thanks, @ayusht2810!

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**